week3OptionalChallenge.js

```diff
@@ -1,53 +1,59 @@
-const findTheMissing = (target) => {
-    // final result list
+function findTheMissing (target){
+   
+    if(typeof target !== "object" || target.length === 0){
+        return 'Invalid input'
+    }
+    else{
     arraySort(target);
-    newArray=[...new Set(target)];
-    let result = [];
+    let newArray=[...new Set(target)];
+    let missingValues = [];
     
     // array will go from min to max value present in the array
     const min = newArray[0];
     const max = newArray[newArray.length - 1];
     
     // will maintain the track of index of newArray array
-    // will start from 2nd element of array because we need a no. to subsctract from
-    let pivotIndex = 1;
+    // will start from 2nd element of array because we need a no. to subtract from
+    let j = 1;
     
-    for (let index = min; index < max; index++) {
+    for (let i = min; i < max; i++) {
     
-        // value to the index
-        let pivotValue = newArray[pivotIndex];
+        let pivotValue = newArray[j];
     
-        // dif of the value
-        let diff = pivotValue - index;
+        let diff = pivotValue - i;
     
```

Comment:
I liked the comments it's better you leave them. Trust me, you may not understand what you were trying to achieve if you come here a few months from now